packages/std/src/addresses.rs

```diff
@@ -86,3 +86,82 @@ impl fmt::Display for CanonicalAddr {
         self.0.fmt(f)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    // Test HumanAddr as_str() for each HumanAddr::from input type
+    #[test]
+    fn human_addr_as_str() {
+        // literal string
+        let human_addr_from_literal_string = HumanAddr::from("literal-string");
+        assert_eq!("literal-string", human_addr_from_literal_string.as_str());
+
+        // String
+        let addr = String::from("Hello, world!");
+        let human_addr_from_string = HumanAddr::from(addr);
+        assert_eq!("Hello, world!", human_addr_from_string.as_str());
+
+        // &HumanAddr
+        let human_addr_from_borrow = HumanAddr::from(&human_addr_from_string);
+        assert_eq!(
+            human_addr_from_borrow.as_str(),
+            human_addr_from_string.as_str()
+        );
+
+        // &&HumanAddr
+        let human_addr_from_borrow_2 = HumanAddr::from(&&human_addr_from_string);
+        assert_eq!(
+            human_addr_from_borrow_2.as_str(),
+            human_addr_from_string.as_str()
+        );
+    }
+
+    #[test]
+    fn human_addr_format() {
+        let human_addr = HumanAddr::from("Hello, world!");
+        assert_eq!("Hello, world!", format!("{}", human_addr));
+    }
+
+    #[test]
+    fn human_addr_len() {
+        let addr = "Hello, world!";
+        let human_addr = HumanAddr::from(addr);
+        assert_eq!(addr.len(), human_addr.len());
+    }
+
+    #[test]
+    fn human_addr_is_empty() {
+        let human_addr = HumanAddr::from("Hello, world!");
+        assert_eq!(false, human_addr.is_empty());
```

Comment:
Can you add a case where `is_empty()` is `true`? It can be here in the same test function

Reply:
Updated :)